router/usecase/pools/routable_balancer_pool.go

```diff
@@ -51,11 +67,23 @@ func (r *routableBalancerPoolImpl) ChargeTakerFeeExactIn(tokenIn sdk.Coin) (toke
 	return tokenInAfterTakerFee
 }
 
+// ChargeTakerFee implements domain.RoutablePool.
+// Charges the taker fee for the given token in and returns the token in after the fee has been charged.
+func (r *routableBalancerPoolImpl) ChargeTakerFeeExactOut(tokenOut sdk.Coin) (tokenOutAfterFee sdk.Coin) {
+	tokenOutAfterTakerFee, _ := poolmanager.CalcTakerFeeExactOut(tokenOut, r.TakerFee)
+	return tokenOutAfterTakerFee
+}
```

Comment:
Why do we need this method?

The quote logic of "out given in" should handle the taker fee charge by charging it on token out

Reply:
Hmm, I must made a mistake when making a notes on my end, because I have noted that Taker fees needed to be calculated based on swap method. I have removed  `ChargeTakerFeeExactOut` methods from all pools.